lib/mix/lib/mix/tasks/format.ex

```diff
@@ -191,6 +253,34 @@ defmodule Mix.Tasks.Format do
       Mix.raise("Expected :subdirectories to return a list of directories, got: #{inspect(subs)}")
     end
 
+    if not is_list(plugins) do
+      Mix.raise("Expected :plugins to return a list of modules, got: #{inspect(plugins)}")
+    end
+
+    if plugins != [] do
+      args = ["--no-elixir-version-check", "--no-deps-check", "--no-archives-check"]
+      Mix.Task.run("loadpaths", args)
+    end
+
+    for plugin <- plugins do
+      cond do
+        not Code.ensure_loaded?(plugin) ->
+          Mix.shell().error(
+            "Skipping formatter plugin #{inspect(plugin)} because module cannot be found"
+          )
+
+        not function_exported?(plugin, :features, 1) ->
+          Mix.shell().error(
+            "Skipping formatter plugin #{inspect(plugin)} because it does not define features/1"
+          )
+
+        true ->
+          :ok
+      end
+    end
+
+    formatter_opts = Keyword.put(formatter_opts, :plugins, plugins)
+
```

Comment:
What do you think about moving all this logic to a new function, like `validate_formatter_opts` or similar?